### DISCUSSION TITLE
Limit switches on auto-squared axes

### DISCUSSION THREAD

**Discussion by Avatario34** [grblHAL/core#261] [Q&A] [answered]
Hi,
I've got my Y- and Z-axis defined as auto-squared. I'm able to see the X-axis limit switch engaging in IOSender and in the GrblHAL realtime status report as well as I can see the Probe input firing. I **can't** see anything for the Y- and Z-axis, neither in the the IOSender GUI, nor in the realtime report. Am I missing a trick? How can I check whether my limit switches work properly on auto-squared axes?
Thanks for your help. 

P.S.: I am using the iMXRT1062 driver on Phil Barrett's GrblHAL breakout board.

**Comment by terjeio** [comment, marked as answer]
What kind of limit switches do you have? Normally open (NO) or Normally closed (NC)?
And is the `$5` setting (Invert limit pins) set accordingly?

> How can I check whether my limit switches work properly on auto-squared axes?

You can get the state of the indiviual limit switches in the console by using the `$lim` command:

```
$lim
[LIMITS:Y,,,]
$lim
[LIMITS:Z,,,]
$lim
[LIMITS:,,Y,]
```

**Reply by Avatario34** [reply]
Thanks, I guess I had a problem with my configuration.
Limit switches are normally open, pins inverted with $5=7.
I'm now getting the following output when I trigger the limit switches in the sequence X, Y1, Y2, Z1, Z2:
```
$lim
[LIMITS:X,,,]
$lim
[LIMITS:,,Y,]
$lim
[LIMITS:Y,,,]
$lim
[LIMITS:Z,,,]
$lim
[LIMITS:,,Z,]
```
So I'm getting what X, Y and Z stand for, but what does the position within the output string tell me (couldn't find a doc for the $lim command)?

**Reply by terjeio** [reply]
> but what does the position within the output string tell me

There can be up to four limit switches per axis in dual configuration, max and max2 is rarely used. min2 and max2 is for the second motor.

`[LIMITS:<min>,<max>,<min2>,<max2>]`

I should add this to the documentation. This, and some other commands, were added for testing code but can be useful for setup too...
`$help commands` list all available...

**Reply by Avatario34** [reply]
That makes sense, thank you!
